### PR TITLE
[MIN-39] Add dvc step to data preparation pipeline

### PR DESCRIPTION
**Pull request by KirsoppJ**
This PR adds a `version_data` step to the data preparation pipeline which can be used to automatically push validated data to the azure storage blob. It also commits and pushes the .dvc files to remote, and tags the commit with the data version name. 

If the `debug_mode` parameter is set to true, nothing is pushed to either the storage bucket or to git and no tag is created. 

To avoid auto-pushes to the develop branch, if the user is on develop, the step throws an error.

### CODE REVIEW

File: pipelines/data_preparation_pipeline/config_data_preparation_pipeline.yaml [hunk 1]
```diff
@@ -5,3 +5,8 @@ steps:
     enable_cache: False
   validate_data:
     enable_cache: False
+  version_data:
+    parameters:
+      data_version_name: "first_version"
```
Comment (osw282) [inline pipelines/data_preparation_pipeline/config_data_preparation_pipeline.yaml:10]: If we run this pipeline again, would this required us to manually change the `data_version_name` in the config?

**Review by osw282 — COMMENTED** [review]
Just one question, looks good overall. 💪🏻

File: pipelines/data_preparation_pipeline/config_data_preparation_pipeline.yaml (hunk 1 above)
Comment (JonoCX) [reply pipelines/data_preparation_pipeline/config_data_preparation_pipeline.yaml:10]: Yep - if you want to use a different data version. If not, then keep it the same.

**Review by JonoCX — APPROVED** [review]
LGTM

**Review by osw282 — APPROVED** [review]
LGTM!

**Pull request merged**